<a href="https://colab.research.google.com/github/bhatimukul/Assignment-func/blob/main/statistics_Advance_2_Assignment_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Q1.Explain the properties of the F-distribution.**

**Ans.** The **F-distribution** is a probability distribution that arises frequently in the analysis of variance (ANOVA), regression analysis, and other statistical tests that compare variances. Below are its key properties:

---

### 1. **Definition**:
The F-distribution is the distribution of the ratio of two independent chi-squared variables divided by their respective degrees of freedom. Mathematically:

\[
F = \frac{\left( \frac{X_1^2}{d_1} \right)}{\left( \frac{X_2^2}{d_2} \right)}
\]

where:
- \( X_1^2 \) and \( X_2^2 \) are independent chi-squared random variables.
- \( d_1 \) and \( d_2 \) are the degrees of freedom of the respective chi-squared variables.

---

### 2. **Shape**:
The shape of the F-distribution is determined by its degrees of freedom:
- \( d_1 \): Degrees of freedom for the numerator.
- \( d_2 \): Degrees of freedom for the denominator.

The shape is skewed to the right, especially for small \( d_1 \) and \( d_2 \), but becomes more symmetric as \( d_1 \) and \( d_2 \) increase.

---

### 3. **Range**:
- The F-distribution is defined for \( F \geq 0 \) because it is based on variances, which cannot be negative.

---

### 4. **Asymmetry**:
- It is positively skewed. The degree of skewness decreases as the degrees of freedom increase.

---

### 5. **Mean**:
The mean of the F-distribution is:
\[
\text{Mean} = \frac{d_2}{d_2 - 2} \quad \text{for } d_2 > 2.
\]
If \( d_2 \leq 2 \), the mean is undefined.

---

### 6. **Variance**:
The variance of the F-distribution is:
\[
\text{Variance} = \frac{2 \cdot d_2^2 \cdot (d_1 + d_2 - 2)}{d_1 \cdot (d_2 - 2)^2 \cdot (d_2 - 4)} \quad \text{for } d_2 > 4.
\]
If \( d_2 \leq 4 \), the variance is undefined.

---

### 7. **Applications**:
The F-distribution is commonly used in:
- **Analysis of Variance (ANOVA)**: To test if the means of multiple groups are equal by comparing group variances.
- **Regression analysis**: To test the overall significance of a model.
- **Equality of variances tests**: Like the Levene's test or Bartlett's test.

---

### 8. **Dependency on Degrees of Freedom**:
- The degrees of freedom \( d_1 \) and \( d_2 \) greatly affect the critical values of the F-distribution.
- The distribution becomes more symmetric as \( d_1 \) and \( d_2 \) increase.

---

### 9. **Relation to Other Distributions**:
- If \( d_1 = 1 \), the F-distribution simplifies to a scaled beta distribution.
- The F-distribution is related to the t-distribution:
  \[
  t^2_{d_2} \sim F_{1, d_2}.
  \]

---

### 10. **Cumulative Distribution Function (CDF)**:
The CDF of the F-distribution is not expressible in a simple closed form but is computed using numerical integration of the incomplete beta function.

---

The F-distribution's properties make it an essential tool for comparing variances and understanding model fit in statistical analysis.

**Q2. In which types of statistical tests is the F-distribution used, and why is it appropriate for these tests?**

**Ans.** The **F-distribution** is used in statistical tests that involve comparing variances or evaluating the significance of models. Its application is grounded in its definition as the ratio of two variances (scaled chi-squared distributions). Below are the primary types of statistical tests where the F-distribution is used and the rationale for its appropriateness:

---

### 1. **Analysis of Variance (ANOVA)**:
- **Purpose**: To compare the means of two or more groups.
- **Why F-distribution is used**:
  - In ANOVA, the test statistic is the ratio of the variance between group means (explained variance) to the variance within groups (unexplained variance).
  - The F-distribution is appropriate because this ratio follows an F-distribution under the null hypothesis that all group means are equal.
- **Example**: Testing whether the average heights of plants under three different fertilizers are equal.

---

### 2. **Regression Analysis**:
- **Purpose**: To test the overall significance of a regression model.
- **Why F-distribution is used**:
  - The F-statistic in regression compares the variance explained by the model (regression sum of squares) to the unexplained variance (error sum of squares).
  - The F-distribution is suitable because this ratio follows the F-distribution if the null hypothesis (no relationship between predictors and outcome) is true.
- **Example**: Testing whether a linear regression model explains a significant amount of variation in sales based on advertising budget.

---

### 3. **Equality of Variances**:
- **Purpose**: To test whether two or more groups have the same variance.
- **Why F-distribution is used**:
  - The test statistic is the ratio of sample variances from two independent groups.
  - Under the null hypothesis that the group variances are equal, this ratio follows an F-distribution.
- **Examples**:
  - **Levene's Test**: For testing equality of variances across multiple groups.
  - **Bartlett’s Test**: Specifically for testing homogeneity of variances under normality.

---

### 4. **Model Comparison (Nested Models)**:
- **Purpose**: To compare two models where one is a simplified version of the other (nested models).
- **Why F-distribution is used**:
  - The F-test assesses whether adding more parameters to a model significantly improves its fit by comparing the variance explained by each model.
  - The test statistic (ratio of explained variances) follows the F-distribution under the null hypothesis that the simpler model is sufficient.
- **Example**: Testing whether adding an interaction term to a regression model improves predictive performance.

---

### 5. **Tests for Specific Hypotheses in Multivariate Analysis**:
- **Purpose**: To evaluate hypotheses in MANOVA, canonical correlation analysis, or similar tests.
- **Why F-distribution is used**:
  - These tests often involve ratios of explained to unexplained variance, similar to ANOVA, and follow the F-distribution under the null hypothesis.
- **Example**: Testing whether the mean vectors of two groups differ across multiple dependent variables.

---

### 6. **Two-Sample Variance Comparison (F-Test)**:
- **Purpose**: To compare the variances of two independent samples.
- **Why F-distribution is used**:
  - The test statistic is the ratio of the sample variances, which follows the F-distribution under the null hypothesis that the variances are equal.
- **Example**: Comparing variability in test scores between two classes.

---

### Why the F-Distribution is Appropriate:
1. **Ratio of Variances**:
   - The F-distribution inherently models ratios of variances, making it ideal for variance-based hypothesis testing.
2. **Skewness**:
   - It accounts for the fact that variances (and their ratios) cannot be negative.
3. **Degrees of Freedom**:
   - Its shape adapts based on the degrees of freedom, reflecting the variability introduced by sample size differences.

The F-distribution's flexibility and theoretical basis make it indispensable for tests involving variance comparisons and model evaluations.

**Q3.What are the key assumptions required for conducting an F-test to compare the variances of two**
**populations?**

**Ans.** The **F-test** for comparing the variances of two populations has specific assumptions that must be met to ensure valid results. These assumptions are as follows:

---

### 1. **Independence**:
- The two samples must be **independent** of each other.
- There should be no relationship between the observations in one sample and those in the other.

---

### 2. **Normality**:
- Both populations should follow a **normal distribution**.
- This assumption is crucial because the F-test is sensitive to deviations from normality. If the data are not normally distributed, the test results may be unreliable.

---

### 3. **Random Sampling**:
- The samples must be drawn from their respective populations using a **random sampling method**.
- This ensures that the samples are representative of the populations.

---

### 4. **Non-Negative Variances**:
- Variances are always non-negative, so the test is only meaningful if both sample variances are greater than zero.

---

### 5. **Equality of Measurement Scale**:
- Both samples must be measured on the same scale (e.g., inches, kilograms) to make their variances comparable.

---

### Considerations for Violations of Assumptions:
1. **Normality**:
   - If the normality assumption is violated, use a non-parametric test like **Levene’s test** or the **Brown-Forsythe test**, which are less sensitive to non-normal distributions.
2. **Independence**:
   - If the samples are not independent, the F-test cannot be used, and alternative methods (e.g., paired tests) should be considered.
3. **Outliers**:
   - Outliers can heavily influence the variance and, consequently, the F-statistic. It's essential to check for and handle outliers appropriately before conducting the test.

---

Meeting these assumptions ensures that the F-test provides accurate and meaningful comparisons of population variances.

**Q4.What is the purpose of ANOVA, and how does it differ from a t-test?**

**Ans.** **Analysis of Variance (ANOVA)** and the **t-test** are both statistical methods used to compare means, but they are suited for different types of problems and have key distinctions. Below is an overview of the purpose of ANOVA and how it differs from the t-test:

---

### **Purpose of ANOVA**
ANOVA is used to determine whether there are significant differences between the means of **three or more groups** by analyzing variances. Its key purposes include:

1. **Assessing Group Differences**:
   - ANOVA tests whether the differences in group means are statistically significant, based on the variance between groups compared to the variance within groups.

2. **Partitioning Variance**:
   - ANOVA decomposes the total variance in the data into two components:
     - **Between-group variance**: Variation due to differences between group means.
     - **Within-group variance**: Variation within each group (random noise).

3. **Hypothesis Testing**:
   - **Null Hypothesis (\(H_0\))**: All group means are equal (\( \mu_1 = \mu_2 = \mu_3 = \dots \)).
   - **Alternative Hypothesis (\(H_a\))**: At least one group mean is different.

4. **Applicability**:
   - Often used in experiments, clinical trials, and observational studies involving multiple groups or factors.

---

### **Differences Between ANOVA and the t-Test**

| **Feature**              | **t-Test**                                | **ANOVA**                                    |
|---------------------------|-------------------------------------------|----------------------------------------------|
| **Purpose**               | Compares means of **two groups**.         | Compares means of **three or more groups**.  |
| **Type of Variance**      | Does not partition variance explicitly.   | Explicitly partitions variance into between-group and within-group components. |
| **Number of Groups**      | Limited to two groups.                   | Can handle three or more groups.             |
| **Types of t-Tests**      | Includes independent and paired t-tests.  | Includes one-way and two-way ANOVA.          |
| **Null Hypothesis (\(H_0\))** | Means of two groups are equal.          | Means of all groups are equal.               |
| **Alternative Hypothesis (\(H_a\))** | Means are different.                  | At least one mean is different.              |
| **Test Statistic**        | Based on the t-distribution.             | Based on the F-distribution.                 |
| **Multiple Groups**       | Requires repeated pairwise comparisons (risk of Type I error inflation). | Evaluates all groups simultaneously without inflating the Type I error. |

---

### **When to Use Each Test**
- **Use a t-test**:
  - When comparing **exactly two groups**.
  - Example: Testing whether the mean test scores of two classes are different.

- **Use ANOVA**:
  - When comparing **three or more groups**.
  - Example: Testing whether three different diets result in different mean weight losses.

---

### **Advantages of ANOVA Over t-Test**
1. **Scalability**:
   - ANOVA handles multiple groups simultaneously, reducing the need for multiple pairwise t-tests.
   
2. **Error Control**:
   - Reduces the risk of **Type I error** inflation, which occurs when performing multiple t-tests.

3. **Versatility**:
   - ANOVA can be extended to include multiple factors (e.g., two-way ANOVA) and interactions.

---

### **Example Comparison**
- **t-Test Example**:
   - Comparing the average heights of males and females.
   - Null Hypothesis: \( \mu_{\text{male}} = \mu_{\text{female}} \).

- **ANOVA Example**:
   - Comparing average heights across three ethnic groups.
   - Null Hypothesis: \( \mu_1 = \mu_2 = \mu_3 \).



**Q5.Explain when and why you would use a one-way ANOVA instead of multiple t-tests when comparing more**
**than two groups.**

**Ans.**You would use a **one-way ANOVA** instead of multiple t-tests when comparing more than two groups for the following reasons:

---

### **When to Use One-Way ANOVA**
1. **Number of Groups**:
   - You have **three or more groups** to compare. For example, you might be comparing the mean test scores of students taught using three different teaching methods.

2. **Single Independent Variable**:
   - There is one independent variable (factor) with multiple levels (categories). For example, "teaching method" with levels: Method A, Method B, and Method C.

3. **Objective**:
   - The goal is to determine whether there is a statistically significant difference in the means of the groups, rather than just comparing groups pair by pair.

---

### **Why Use One-Way ANOVA Instead of Multiple t-Tests**

#### 1. **Error Rate Control**:
- **Problem with Multiple t-Tests**:
   - Conducting multiple t-tests increases the likelihood of a **Type I error** (false positive). The probability of at least one Type I error increases with the number of t-tests.
   - For \( n \) groups, the number of pairwise comparisons is \( \frac{n(n-1)}{2} \). For 4 groups, this requires 6 t-tests; for 5 groups, 10 t-tests, and so on.

   \[
   \text{Family-wise error rate (FWER)} = 1 - (1 - \alpha)^k
   \]
   where \( \alpha \) is the significance level (e.g., 0.05) and \( k \) is the number of comparisons.

- **Solution with ANOVA**:
   - One-way ANOVA performs a **single test** to evaluate differences across all groups simultaneously, maintaining the overall significance level (\( \alpha \)).

---

#### 2. **Simplicity and Efficiency**:
- Conducting multiple t-tests is cumbersome and inefficient as the number of groups increases. One-way ANOVA provides a single test for evaluating all group differences at once.

---

#### 3. **Comprehensive Analysis**:
- **One-way ANOVA** evaluates the overall variance among all groups (using between-group and within-group variance). If the null hypothesis is rejected, it suggests at least one group mean is different but does not indicate which groups differ. Post-hoc tests can then be used for specific group comparisons.

---

#### 4. **Theoretical Foundation**:
- The F-statistic used in one-way ANOVA is derived from a ratio of variances, which provides a robust measure of overall group differences under the assumption of normality and equal variances.

---

### **Example Comparison**

**Scenario**: Testing whether three teaching methods (A, B, and C) result in different mean test scores.

- **Multiple t-tests**:
  - You would perform 3 pairwise tests: A vs. B, B vs. C, and A vs. C.
  - Risk: Increased Type I error due to repeated testing.

- **One-way ANOVA**:
  - A single test evaluates whether the means of all three groups are significantly different.
  - Null Hypothesis (\( H_0 \)): The means of all groups are equal (\( \mu_A = \mu_B = \mu_C \)).
  - Alternative Hypothesis (\( H_a \)): At least one group mean is different.

**Q6.Explain how variance is partitioned in ANOVA into between-group variance and within-group variance.**
**How does this partitioning contribute to the calculation of the F-statistic?**

**Ans.** In **Analysis of Variance (ANOVA)**, the total variability in the data is partitioned into two components: **between-group variance** and **within-group variance**. This partitioning is central to the calculation of the **F-statistic**, which determines whether there are significant differences between group means.

---

### **Partitioning of Variance in ANOVA**

#### 1. **Total Variance**:
The total variance in the data measures the overall variability of all observations from the grand mean. It is quantified by the **Total Sum of Squares (SST)**:
\[
\text{SST} = \sum_{i=1}^{k} \sum_{j=1}^{n_i} (Y_{ij} - \bar{Y})^2
\]
where:
- \( Y_{ij} \): The \( j \)-th observation in the \( i \)-th group.
- \( \bar{Y} \): The grand mean (mean of all observations across groups).
- \( k \): Number of groups.
- \( n_i \): Number of observations in group \( i \).

#### 2. **Between-Group Variance**:
This represents the variability due to differences between group means and is measured by the **Between-Group Sum of Squares (SSB)**:
\[
\text{SSB} = \sum_{i=1}^{k} n_i (\bar{Y}_i - \bar{Y})^2
\]
where:
- \( \bar{Y}_i \): Mean of the \( i \)-th group.
- \( n_i \): Number of observations in the \( i \)-th group.

It reflects how far each group mean (\( \bar{Y}_i \)) is from the grand mean (\( \bar{Y} \)).

#### 3. **Within-Group Variance**:
This represents the variability within each group and is measured by the **Within-Group Sum of Squares (SSW)**:
\[
\text{SSW} = \sum_{i=1}^{k} \sum_{j=1}^{n_i} (Y_{ij} - \bar{Y}_i)^2
\]
It captures the deviations of individual observations from their group mean (\( \bar{Y}_i \)).

---

### **Relationship Between Variance Components**:
The total variance is the sum of between-group variance and within-group variance:
\[
\text{SST} = \text{SSB} + \text{SSW}
\]

---

### **Contribution to the F-Statistic**

The **F-statistic** in ANOVA is calculated as the ratio of the between-group variance to the within-group variance. This ratio assesses whether the variation between groups is significantly larger than the variation within groups, which would suggest a difference in group means.

#### 1. **Mean Squares**:
- **Between-Group Mean Square (MSB)**:
  \[
  \text{MSB} = \frac{\text{SSB}}{k-1}
  \]
  where \( k-1 \) is the degrees of freedom for the between-group variability.
  
- **Within-Group Mean Square (MSW)**:
  \[
  \text{MSW} = \frac{\text{SSW}}{N-k}
  \]
  where \( N-k \) is the degrees of freedom for the within-group variability, with \( N \) being the total number of observations.

#### 2. **F-Statistic**:
The F-statistic is the ratio of these two mean squares:
\[
F = \frac{\text{MSB}}{\text{MSW}}
\]

- A **large F-value** suggests that the between-group variance is much greater than the within-group variance, indicating significant differences among group means.
- A **small F-value** suggests that the between-group variance is comparable to or smaller than the within-group variance, indicating no significant differences among group means.

---

### **Summary of Variance Partitioning and the F-Statistic**
1. The **total variance** (SST) is split into:
   - **Between-group variance (SSB)**: Variability due to differences between group means.
   - **Within-group variance (SSW)**: Variability due to differences within groups.

2. These variances are standardized into mean squares (MSB and MSW).

3. The **F-statistic** (\( F = \text{MSB} / \text{MSW} \)) compares the two, testing whether the variability between groups is significantly larger than the variability within groups.

This partitioning allows ANOVA to test the hypothesis that group means are equal in a systematic and interpretable way.

**Q7.Compare the classical (frequentist) approach to ANOVA with the Bayesian approach. What are the key**
d**ifferences in terms of how they handle uncertainty, parameter estimation, and hypothesis testing?**

Ans.The **classical (frequentist) approach** to ANOVA and the **Bayesian approach** to ANOVA differ fundamentally in how they conceptualize and handle uncertainty, parameter estimation, and hypothesis testing. Below is a comparison of the two approaches:

---

### **1. Uncertainty**
- **Frequentist Approach**:
  - Views probability as the long-run frequency of events.
  - Uncertainty is quantified through **p-values** and confidence intervals, which rely on repeated sampling concepts.
  - Assumes fixed parameters (e.g., group means and variances) and tests hypotheses about their values using observed data.

- **Bayesian Approach**:
  - Views probability as a degree of belief about a parameter, updated with data.
  - Uncertainty is expressed using **posterior distributions**, which combine prior knowledge (or beliefs) with observed data via Bayes’ theorem.
  - Parameters are treated as random variables with distributions that reflect uncertainty about their true values.

---

### **2. Parameter Estimation**
- **Frequentist Approach**:
  - Estimates parameters (e.g., group means and variances) using **point estimates** such as sample means and variances.
  - No prior knowledge about the parameters is incorporated; estimation is based solely on the data.
  - Does not provide direct probabilities about parameter values (e.g., cannot say "the mean is likely between X and Y").

- **Bayesian Approach**:
  - Estimates parameters by generating **posterior distributions** that reflect uncertainty about parameter values.
  - Combines prior distributions (representing prior beliefs) with likelihood derived from the observed data.
  - Allows probabilistic statements, such as "there is a 95% probability that the mean is between X and Y."

---

### **3. Hypothesis Testing**
- **Frequentist Approach**:
  - Tests the null hypothesis (\(H_0\)) that all group means are equal using an F-statistic:
    \[
    F = \frac{\text{Between-group variance}}{\text{Within-group variance}}
    \]
  - A small **p-value** (e.g., \(p < 0.05\)) leads to rejection of \(H_0\), but it does not quantify the probability that \(H_0\) is true.
  - Results depend on a fixed significance level (\(\alpha\)) and do not incorporate prior information.

- **Bayesian Approach**:
  - Tests hypotheses by comparing **posterior probabilities** or computing **Bayes factors**, which quantify the strength of evidence for competing models (e.g., equal vs. unequal means).
    \[
    \text{Bayes Factor} = \frac{\text{Probability of Data under Model 1}}{\text{Probability of Data under Model 2}}
    \]
  - Allows direct probabilities about hypotheses, such as "there is a 90% probability that the means differ."
  - Naturally incorporates prior beliefs about hypotheses or parameters.

---

### **4. Interpretation of Results**
- **Frequentist Approach**:
  - A significant result (small p-value) indicates that the data are unlikely under the null hypothesis, leading to its rejection.
  - Provides no direct measure of the probability of the null hypothesis or alternative hypothesis being true.
  - Confidence intervals indicate a range of plausible values for parameters but do not have a probabilistic interpretation.

- **Bayesian Approach**:
  - Posterior probabilities directly express the degree of belief in a hypothesis or parameter value.
  - Results are probabilistic and provide more intuitive interpretations (e.g., "there is an 80% probability that the difference between means exceeds 5").

---

### **5. Prior Information**
- **Frequentist Approach**:
  - Does not incorporate prior information; conclusions are based entirely on the observed data.
  - Results are objective in the sense that they depend only on the data and the chosen significance level.

- **Bayesian Approach**:
  - Explicitly incorporates prior information through prior distributions.
  - Results can vary depending on the chosen priors, which can be subjective or objective.

---

### **6. Computational Requirements**
- **Frequentist Approach**:
  - Relies on relatively simple calculations (e.g., F-statistic, p-values) and assumes analytical solutions.
  - Computationally less intensive, especially for simple designs.

- **Bayesian Approach**:
  - Often requires computationally intensive methods, such as **Markov Chain Monte Carlo (MCMC)**, to approximate posterior distributions.
  - More demanding computationally, particularly for complex models.

---

### **7. Applications**
- **Frequentist Approach**:
  - Standard method for hypothesis testing in many fields, especially when prior information is unavailable or undesirable.
  - Commonly used in situations where simplicity and transparency are prioritized.

- **Bayesian Approach**:
  - Preferred when prior information is available and relevant, or when a probabilistic interpretation of results is desired.
  - Useful for small-sample studies, hierarchical models, and complex experimental designs.

---

### **Key Differences at a Glance**

| **Aspect**              | **Frequentist ANOVA**                           | **Bayesian ANOVA**                              |
|--------------------------|------------------------------------------------|------------------------------------------------|
| **Uncertainty**          | Based on long-run frequencies (p-values).      | Based on posterior distributions.              |
| **Parameter Estimates**  | Point estimates (e.g., means, variances).      | Posterior distributions reflecting uncertainty. |
| **Hypothesis Testing**   | Uses p-values to reject \(H_0\).               | Computes posterior probabilities or Bayes factors. |
| **Incorporates Priors**  | No.                                             | Yes, explicitly.                               |
| **Interpretation**       | Indirect (reject \(H_0\) if p-value is small). | Direct (probabilities about hypotheses).       |
| **Computation**          | Simple and fast.                               | Computationally intensive (e.g., MCMC).        |

---



**Q8.Question: You have two sets of data representing the incomes of two different professions:**

. Profession A: [48, 52, 55, 60, 62'

. Profession B: [45, 50, 55, 52, 47] Perform an F-test to determine if the variances of the two professions'
incomes are equal. What are your conclusions based on the F-test?

Task: Use Python to calculate the F-statistic and p-value for the given data

Objective: Gain experience in performing F-tests and interpreting the results in terms of variance comparison.

In [1]:
import numpy as np
from scipy import stats

# Data for both professions
profession_A = [48, 52, 55, 60, 62]
profession_B = [45, 50, 55, 52, 47]

# Calculate variances for both professions
var_A = np.var(profession_A, ddof=1)  # Sample variance
var_B = np.var(profession_B, ddof=1)  # Sample variance

# Calculate the F-statistic (larger variance / smaller variance)
F_statistic = var_A / var_B if var_A > var_B else var_B / var_A

# Degrees of freedom for both groups
df_A = len(profession_A) - 1
df_B = len(profession_B) - 1

# Calculate the p-value using the F-distribution
p_value = 1 - stats.f.cdf(F_statistic, df_A, df_B)

F_statistic, p_value


(2.089171974522293, 0.24652429950266952)

The F-statistic for comparing the variances of the incomes of the two professions is approximately **2.09**, and the corresponding **p-value** is approximately **0.247**.

### Interpretation:
- The **null hypothesis** (\(H_0\)) for the F-test is that the variances of the two populations are equal.
- The **alternative hypothesis** (\(H_a\)) is that the variances are not equal.

Given that the p-value (\(0.247\)) is greater than the commonly used significance level of **0.05**, we **fail to reject the null hypothesis**. This means there is not enough evidence to conclude that the variances of the two professions' incomes are significantly different.

### Conclusion:
Based on the F-test, we conclude that there is no significant difference in the variances of the incomes between Profession A and Profession B.

Q**9.Question: Conduct a one-way ANOVA to test whether there are any statistically significant differences in**
**average heights between three different regions with the following data1**

. Region A: [160, 162, 165, 158, 164'

. Region B: [172, 175, 170, 168, 174'

. Region C: [180, 182, 179, 185, 183'

. Task: Write Python code to perform the one-way ANOVA and interpret the results

. Objective: Learn how to perform one-way ANOVA using Python and interpret F-statistic and p-value

In [2]:
from scipy.stats import f_oneway

# Data for the three regions
region_A = [160, 162, 165, 158, 164]
region_B = [172, 175, 170, 168, 174]
region_C = [180, 182, 179, 185, 183]

# Perform one-way ANOVA
F_statistic, p_value = f_oneway(region_A, region_B, region_C)

F_statistic, p_value


(67.87330316742101, 2.870664187937026e-07)

The results of the one-way ANOVA are as follows:

- **F-statistic**: 67.87
- **p-value**: \(2.87 \times 10^{-7}\)

### Interpretation:
- The **null hypothesis** (\(H_0\)) is that the average heights across the three regions are equal.
- The **alternative hypothesis** (\(H_a\)) is that at least one region has a different average height.

Since the p-value (\(2.87 \times 10^{-7}\)) is much smaller than the common significance level (\( \alpha = 0.05 \)), we **reject the null hypothesis**.

### Conclusion:
There is strong evidence to conclude that there are statistically significant differences in the average heights between the three regions. Post-hoc tests (e.g., Tukey's HSD) can be performed to identify which specific regions differ in their average heights.